From da4fe5dbf9c22199639a8cb1d863ee7271eb8077 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Mon, 27 Aug 2018 09:41:06 -0700
Subject: [PATCH] TRITON-724 sdcadm's flushLogs can fail to callback resulting
 in exit status 0 when there is an error to report

---
 lib/cli/index.js |  6 ++---
 lib/logging.js   | 61 ++----------------------------------------------
 2 files changed, 4 insertions(+), 63 deletions(-)

diff --git a/lib/cli/index.js b/lib/cli/index.js
index cdf896a..f7cb0cf 100644
--- a/lib/cli/index.js
+++ b/lib/cli/index.js
@@ -223,11 +223,9 @@ CLI.prototype.fini = function fini(subcmd, err, cb) {
         }
         this.log[logLevel]({subcmd: subcmd, exitStatus: exitStatus, cli: true},
             'cli exit');
-
-        logging.flushLogs([this.log], cb);
-    } else {
-        cb();
     }
+
+    cb();
 };
 
 CLI.prototype.do_self_update = require('./do_self_update').do_self_update;
diff --git a/lib/logging.js b/lib/logging.js
index 72ebdd6..8a14a09 100644
--- a/lib/logging.js
+++ b/lib/logging.js
@@ -161,7 +161,7 @@ OpenOnErrorFileStream.prototype.write1 = function (rec) {
 // Used when writing to file.
 OpenOnErrorFileStream.prototype.write2 = function (rec) {
     var str = JSON.stringify(rec, bunyan.safeCycles()) + '\n';
-    this.stream.write(str);
+    return this.stream.write(str);
 };
 
 
@@ -221,63 +221,6 @@ function createLogger(opts) {
 
 
 
-/**
- * Flush all open log streams
- */
-function flushLogs(logs, callback) {
-    if (!logs) {
-        callback();
-        return;
-    }
-
-    var streams = [];
-    if (!util.isArray(logs)) {
-        logs = [ logs ];
-    }
-
-    if (logs.length === 0) {
-        callback();
-        return;
-    }
-
-    logs.forEach(function (log) {
-        if (!log || !log.streams || log.streams.length === 0) {
-            return;
-        }
-
-        streams = streams.concat(log.streams);
-    });
-
-    var toClose = streams.length;
-    var closed = 0;
-
-    function _doneClose() {
-        closed++;
-        if (closed === toClose) {
-            callback();
-            return;
-        }
-    }
-
-    streams.forEach(function (str) {
-        if (!str || !str.stream) {
-            _doneClose();
-            return;
-        }
-
-        str.stream.once('drain', function () {
-            _doneClose();
-        });
-
-        if (str.stream.write('')) {
-            _doneClose();
-        }
-    });
-}
-
-
-
 module.exports = {
-    createLogger: createLogger,
-    flushLogs: flushLogs
+    createLogger: createLogger
 };
-- 
2.21.0

